iceoryx_utils/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -370,29 +383,48 @@ cxx::expected<int32_t, IpcChannelError> UnixDomainSocket::createSocket(const Ipc
         }
         else
         {
+            auto closeCall = closeFd(sockfd);
+            if (closeCall.has_error())
+            {
+                return cxx::error<IpcChannelError>(closeCall.get_error());
+            }
             return createErrorFromErrnum(bindCall.getErrNum());
         }
     }
     else
     {
-        // we use a connected socket, this leads to a behavior closer to the message queue (e.g. error if client is
-        // created and server not present)
+        // we use a connected socket, this leads to a behavior closer to the message queue (e.g. error if client
+        // is created and server not present)
         auto connectCall = cxx::makeSmartC(connect,
                                            cxx::ReturnMode::PRE_DEFINED_ERROR_CODE,
                                            {ERROR_CODE},
-                                           {},
+                                           {ENOENT},
                                            sockfd,
                                            (struct sockaddr*)&m_sockAddr,
                                            static_cast<socklen_t>(sizeof(m_sockAddr)));
 
-        if (!connectCall.hasErrors())
+        if (connectCall.hasErrors())
         {
-            return cxx::success<int32_t>(sockfd);
+            auto closeCall = closeFd(sockfd);
+            if (closeCall.has_error())
+            {
+                return cxx::error<IpcChannelError>(closeCall.get_error());
+            }
+            return createErrorFromErrnum(connectCall.getErrNum());
         }
-        else
+        else if (connectCall.getErrNum() == ENOENT)
```

Comment:
This is an error/sad path but `hasErrors()` didn't return true, when taking this branch because `ENOENT` shall not be blamed as a error inside `SmartC`. This could be a good first issue, I've added it to #418

Reply:
maybe we need something like silent errors in SmartC for stuff like this